src/key_info_managers/mod.rs

```diff
@@ -146,60 +146,82 @@ pub trait ManageKeyInfo {
+    ///
+    /// # Errors
+    ///
+    /// Returns an error as a String if there was a problem accessing the Key Info Manager.
+    pub fn list_keys(
+        &self,
+        app_name: &ApplicationName,
+        provider_id: ProviderID,
+    ) -> Result<Vec<parsec_interface::operations::list_keys::KeyInfo>, String> {
+        use parsec_interface::operations::list_keys::KeyInfo;
+
+        let mut keys: Vec<KeyInfo> = Vec::new();
+        let key_triples = self.get_all(provider_id)?;
+
+        for key_triple in key_triples {
+            if key_triple.app_name() != app_name {
+                continue;
+            }
+
+            let key_info = self.get(key_triple)?;
+            let key_info = match key_info {
+                Some(key_info) => key_info,
+                _ => continue,
+            };
+
+            keys.push(KeyInfo {
+                provider_id: key_triple.provider_id,
+                name: key_triple.key_name().to_string(),
+                attributes: key_info.attributes,
+            });
         }
 
-        let key_info = manager.get(key_triple)?;
-        let key_info = match key_info {
-            Some(key_info) => key_info,
-            _ => continue,
-        };
-
-        keys.push(KeyInfo {
-            provider_id: key_triple.provider_id,
-            name: key_triple.key_name().to_string(),
-            attributes: key_info.attributes,
-        });
+        Ok(keys)
     }
 
-    Ok(keys)
-}
+    /// Returns a Vec of ApplicationName of clients having keys in the provider.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error as a String if there was a problem accessing the Key Info Manager.
+    pub fn list_clients(&self, provider_id: ProviderID) -> Result<Vec<ApplicationName>, String> {
+        let key_triples = self.get_all(provider_id)?;
+        let mut clients = Vec::new();
+
+        for key_triple in key_triples {
+            if !clients.contains(key_triple.app_name()) {
+                let _ = clients.push(key_triple.app_name().clone());
+            }
+        }
 
-/// Returns a Vec of ApplicationName of clients having keys in the provider.
-///
-/// # Errors
-///
-/// Returns an error as a String if there was a problem accessing the Key Info Manager.
-pub fn list_clients(
-    manager: &dyn ManageKeyInfo,
-    provider_id: ProviderID,
-) -> Result<Vec<ApplicationName>, String> {
-    let key_triples = manager.get_all(provider_id)?;
-    let mut clients = Vec::new();
+        Ok(clients)
+    }
 
-    for key_triple in key_triples {
-        if !clients.contains(key_triple.app_name()) {
-            let _ = clients.push(key_triple.app_name().clone());
+    /// Check if a key triple exists in the Key Info Manager and return a ResponseStatus
+    ///
+    /// # Errors
+    ///
+    /// Returns PsaErrorAlreadyExists if the key triple already exists or KeyInfoManagerError for
+    /// another error.
+    pub fn already_exists(&self, key_triple: &KeyTriple) -> Result<(), ResponseStatus> {
```

Comment:
I get that the doc above says how this method works, but to me `already_exists` implies that `Ok(())` would mean `yes`, not `no` - at least that's what I'd assume it does if I see it used somewhere in a PR. Maybe something like `is_unique` (or `check_unique`) matches the return type better? Or return `Result<bool, ResponseStatus>`?

Reply:
ah yes you are correct, now that I look at it it's kind of wrong 🤡 
I changed the return type following the same kind of convention we use in the interface:
* a method starting by `is` (an interrogation) return just a `bool`
* a method which name is a declaration returns a `Result`

I can change the name to `does_not_already_exist`